code/modules/research/mechanic/mechanic_fabs.dm

```diff
@@ -183,3 +183,10 @@
 		else
 			return 0
 	return 0
+
+/obj/machinery/r_n_d/fabricator/mechanic_fab/after_craft(datum/design/part)
+	if(!being_built || !being_built.reagents)
```

Comment:
I don't think you meant to make it work this way: the proc assumes that if the `being_built` object has no reagents, then there's no need to look for other objects that contain reagents.
On that note, I'm not sure if simply searching `being_built`'s contents for objects that have reagents is correct either:
- It could be not sufficient if the reagents are nested more deeply (you might want to search recursively)
- It could be wrong if some fab-able object *does* need to spawn with reagents, though I can't think of one right now. This second one might not be a concern.

Reply:
Yes that's a good point, I actually thought of the exact same thing while I was out today, a recursive search is much better. Although probably unlikely to come up, it's safer.